charts/k6-operator/templates/prometheus/serviceMonitor.yaml

```diff
@@ -12,8 +12,11 @@ metadata:
   annotations:
     {{- include "k6-operator.customAnnotations" . | default "" | nindent 4 }}
   name: controller-manager-metrics-monitor
-  namespace: system
+  namespace: {{- include "k6-operator.namespace" . }}
 spec:
+  namespaceSelector:
```

Comment:
It [seems like](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#namespace-selector) `namespaceSelector` requires 1.24+ ? We're "officially" 1.23+ now (can be seen [here](https://grafana.com/docs/grafana-cloud/k6/author-run/private-load-zone-v2/#before-you-begin)). I think it'd be nice to increase this version some time ~ next year though, and then this addition would be great.

Reply:
`namespaceSelector` is part of the [ServiceMonitor CRD](https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/api.md#servicemonitor) and not part of native k8s semantics. From my perspective, whether the field is usable in an earlier version of k8s is probably an issue for the prometheus operator (or any operator which implements the CRD) and not the concern here.

Reply:
@edwintye is right, the above [link](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#namespace-selector) is specific to Pods, not ServiceMonitors.

Reply:
Thanks for clarification and the link! Agreed with the logic of this being the issue at the level of Prometheus operator :+1: